src/schema/sources.ts

```diff
@@ -486,25 +495,34 @@ export const canAccessSource = async (
 
   const sourceId = source.id;
   const repo = ctx.getRepository(SourceMember);
-  const [loggedUser, validateRankAgainst] = await Promise.all([
-    repo.findOneBy({ sourceId, userId: ctx.userId }),
-    requireGreaterAccessPrivilege[permission]
-      ? repo.findOneByOrFail({ sourceId, userId: validateRankAgainstId })
-      : Promise.resolve(null),
-  ]);
-
-  if (!loggedUser) {
+  const validateRankAgainst = await (requireGreaterAccessPrivilege[permission]
+    ? repo.findOneByOrFail({ sourceId, userId: validateRankAgainstId })
+    : Promise.resolve(null));
+
+  if (!sourceMember) {
     return false;
   }
 
   return hasPermissionCheck(
     source,
-    loggedUser,
+    sourceMember,
     permission,
     validateRankAgainst,
   );
 };
 
+export const canPostToSquad = (
+  ctx: Context,
+  squad: SquadSource,
+  sourceMember: SourceMember,
+): boolean => {
+  if (!sourceMember) {
+    return false;
+  }
+
+  return sourceRoleRank[sourceMember.role] >= squad.memberPostingRank;
```

Comment:
Are you sure we can't use the source permission check for this? 👀

Reply:
Agreed in slack that this is ok for now. We might need to streamline permissions handling long term since we have multiple functions doing similar things.